services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -637,21 +669,144 @@ srcLldpDevices=#newSrcEnabled, dstLldpDevices=#newDstEnabled"() {
+        flowHelper.addFlow(flow)
+
+        when: "Two devices send lldp packet on each flow endpoint"
+        def srcData = LldpData.buildRandom()
+        def dstData = LldpData.buildRandom()
+        def nonDefaultVlan = 777 //use this vlan if flow endpoint is 'default' and should catch any vlan
+        def tgService = traffExamProvider.get()
+        withPool {
+            [[flow.source, srcData], [flow.destination, dstData]].eachParallel { endpoint, lldpData ->
+                new ConnectedDevice(tgService, topology.getTraffGen(endpoint.datapath),
+                        endpoint.vlanId ?: nonDefaultVlan).withCloseable {
+                    it.sendLldp(lldpData)
+                        }
+            }
+        }
+
+        then: "Both device are registered for the flow on src and dst"
+        Wrappers.wait(WAIT_OFFSET) {
+            with(northbound.getFlowConnectedDevices(flow.id)) {
+                it.source.lldp.size() == 1
+                it.destination.lldp.size() == 1
+                verifyEquals(it.source.lldp.first(), srcData)
+                verifyEquals(it.destination.lldp.first(), dstData)
+            }
+        }
+
+        and: "Device is registered on src switch"
+        with(northboundV2.getConnectedDevices(flow.source.datapath).ports) {
+            it.size() == 1
+            it[0].portNumber == srcTg.switchPort
+            it[0].lldp.size() == 1
+            it[0].lldp.first().flowId == flow.id
+            it[0].lldp.first().vlan == (flow.source.vlanId ?: nonDefaultVlan)
+            verifyEquals(it[0].lldp.first(), srcData)
+        }
+
+        and: "Device is registered on dst switch"
+        with(northboundV2.getConnectedDevices(flow.destination.datapath).ports) {
+            it.size() == 1
+            it[0].portNumber == dstTg.switchPort
+            it[0].lldp.size() == 1
+            it[0].lldp.first().flowId == flow.id
+            it[0].lldp.first().vlan == (flow.destination.vlanId ?: nonDefaultVlan)
+            verifyEquals(it[0].lldp.first(), dstData)
+        }
+
+        cleanup: "Cleanup: delete the flow"
+        flowHelperV2.deleteFlow(flow.id)
+
+        and: "Restore initial switch properties"
+        restoreSwitchProperties(flow.source.datapath, initialSrcProps)
+        restoreSwitchProperties(flow.destination.datapath, initialDstProps)
+        [flow.source.datapath, flow.destination.datapath].each { database.removeConnectedDevices(it) }
+
+        and: "Cleanup LLDP meters because feature https://github.com/telstra/open-kilda/issues/2969 is not implemented yet"
+        cleanupLldpMeters(flow.source.datapath)
+        cleanupLldpMeters(flow.destination.datapath)
+
+        where:
+        srcDefault | dstDefault
+        false      | true
+        true       | true
+
+        descr = "default(no-vlan) flow endpoints on ${getDescr(srcDefault, dstDefault)}"
     }
 
     def "System forbids to turn on 'connected devices per switch' on a single-table-mode switch"() {
```

Comment:
Does system allow to disable `multiTableMode` when `lldp` is enabled?

Reply:
Nice catch, let me try